File: Valid Palindrome/Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        std::vector<char>   str;
        int diff = 'a' - 'A';
        for(unsigned int i = 0; i < s.size(); ++i){
            char c = s[i];
            if((c >= 'A') && (c <= 'Z')){
                c += diff;
                str.push_back(c);
            }
            else if((c >= 'a') && (c <= 'z')){
                str.push_back(c);
            }
            else if((c >= '0') && (c <= '9')){
                str.push_back(c);
            }
            else{
                // Ignore
            }
        }
        for(unsigned int i = 0; i < str.size() / 2; ++i){
            if(str[i] != str[str.size() - 1 - i])       return false;
        }
        return true;
    }
};

File: Valid Palindrome/Valid Palindrome.java
public class Solution {
    public boolean isPalindrome(String s) {
        char[] str = s.toLowerCase().toCharArray();
        int front = 0;
        int end = str.length - 1;
        while(front < end){
            while(front < str.length && !((str[front] >= 'a' && str[front] <= 'z') || (str[front] >= '0' && str[front] <= '9')))     ++front;
            if(front > end)     return true;
            while(end >= 0 && !((str[end] >= 'a' && str[end] <= 'z') || (str[end] >= '0' && str[end] <= '9')))     --end;
            if(front > end)     return true;
            if(str[front] != str[end])          return false;
            ++front;
            --end;
        }
        return true;
    }
}
